cpp/include/cudf/table/row_operators.cuh

```diff
@@ -222,13 +223,75 @@ class element_equality_comparator {
   bool nulls_are_equal;
 };
 
+namespace detail {
+
+/**
+ * @brief Helper functor to check if two types `LhsType` and `RhsType` are comparable for equality.
+ */
+template <typename LhsType>
+struct equality_comparable_functor {
+  template <typename RhsType>
+  bool __device__ operator()() const
+  {
+    return cudf::is_equality_comparable<LhsType, RhsType>();
+  }
+};
+
+/**
+ * @brief Helper functor to check if a specified column `rhs_col` can be compared for equality
+ * with type `LhsType`.
+ */
+struct lhs_equality_comparable_functor {
+  template <typename LhsType>
+  bool __device__ operator()(column_device_view rhs_col) const
+  {
+    return cudf::type_dispatcher(rhs_col.type(), equality_comparable_functor<LhsType>{});
+  }
+};
+
+/**
+ * @brief Checks if two columns have types that permit equality comparisons
+ *
+ * @param lhs Left column for equality comparisons
+ * @param rhs Right column for equality comparisons
+ * @return true If `lhs` and `rhs` columns have types whose values may be compared for equality
+ * @return false Otherwise.
+ */
+bool __device__ columns_are_equality_comparable(column_device_view lhs, column_device_view rhs)
```

Comment:
I don't think this should be a device function.

Reply:
But isn't `table_device_view::column(size_type)` a `__device__` function?